mypy/test/testcheck.py

```diff
@@ -167,6 +176,9 @@ def run_case_once(
             # Not incremental
             msg = "Unexpected type checker output ({}, line {})"
             output = testcase.output
+            output = sorted(
+                output, key=lambda s: s.partition(":")[0], reverse=True
+            )  # stable-sort by filename
```

Comment:
Not sure how we did without this before. This is basically so that expected errors extracted from `[case ...]` and `[file ...]` sections would match mypy's output; i.e. to compensate for the order of files not matching. (The order of lines is still expected to match.)

Maybe in all those cases we had to switch to `[out]` kind of assertions?

Reply:
I should break it out into a separate PR.

Reply:
👉 https://github.com/python/mypy/pull/15290